cpp/INIReader.cpp

```diff
@@ -68,6 +68,14 @@ bool INIReader::GetBoolean(const string& section, const string& name, bool defau
         return default_value;
 }
 
+// Return true if the given section exists
+bool INIReader::HasSection(const std::string& section) const
+{
+    const string test = MakeKey(section, "");
+    return std::any_of(_values.begin(), _values.end(),
+        [&](const std::pair<string, string>& k){return k.first.find(test) == 0;});
```

Comment:
I believe `any_of` loops through all values in the map. This is an ordered map, so we should be able to to a binary search for the position and see if a value with that section exists. How about something like this (not extensively tested):

```c++
bool INIReader::HasSection(const string& section) const
{
    string key = MakeKey(section, "");
    std::map<string, string>::const_iterator pos = _values.lower_bound(key);
    if (pos == _values.end())
        return false;
    // Does the key at the lower_bound pos start with "section"?
    return pos->first.compare(0, key.length(), key) == 0;
}
```

Reply:
Yep, that's better. I updated the PR.